src/ServiceLayer.Mesh/FileTypes/NbssAppointmentEvents/StagingPersister.cs

```diff
@@ -1,13 +1,53 @@
+using System.Globalization;
+using ServiceLayer.Data;
+using ServiceLayer.Data.Models;
 using ServiceLayer.Mesh.FileTypes.NbssAppointmentEvents.Models;
 
 namespace ServiceLayer.Mesh.FileTypes.NbssAppointmentEvents;
 
-// TODO - class to take validated AppointmentEventsFile and save the records to NbssAppointmentEvents table
-public class StagingPersister : IStagingPersister
+public class StagingPersister(ServiceLayerDbContext dbContext) : IStagingPersister
 {
-    public Task WriteStagedData(ParsedFile parsedFile)
+    public async Task WriteStagedData(ParsedFile parsedFile, MeshFile meshFile)
     {
-        // TODO - implement this
-        throw new NotImplementedException();
+        var nbssAppointmentEvents = MapFileDataRecordsToNbssAppointmentEvents(parsedFile, meshFile.FileId);
+
+        await dbContext.NbssAppointmentEvents.AddRangeAsync(nbssAppointmentEvents);
+        await dbContext.SaveChangesAsync();
+    }
+
+    private static List<NbssAppointmentEvent> MapFileDataRecordsToNbssAppointmentEvents(ParsedFile parsedFile, string fileId)
+    {
+        return [.. parsedFile.DataRecords.Select(record => new NbssAppointmentEvent
+        {
+            MeshFileId = fileId,
+            BSO = record.Fields["BSO"],
+            ExtractId = parsedFile.FileHeader!.ExtractId!,
+            Sequence = record.Fields["Sequence"],
+            Action = record.Fields["Action"],
+            ClinicCode = record.Fields["Clinic Code"],
+            HoldingClinic = record.Fields["Holding Clinic"],
```

Comment:
If this input field is blank (could be empty string or blank, it's not clear), map it to a null.

There are a couple of places we want to do this, so suggest making a helper method

```
public static string NullIfWhiteSpace(string input) =>
    string.IsNullOrWhiteSpace(input) ? null : input;
```

then this line becomes

```
HoldingClinic = NullIfWhiteSpace(record.Fields["Holding Clinic"]),
```